docs/INSTALL-devel.rst

```diff
@@ -0,0 +1,292 @@
+==================
+
+On systems with older versions of ``pip``, such as Debian Wheezy and Ubuntu
+12.04, this install will fail with the following error::
+
+  AttributeError: 'NoneType' object has no attribute 'skip_requirements_regex'
+
+Fix this by upgrading ``pip`` within the virtual environment::
+
+  pip install --upgrade pip
+
+Versions of python prior to 2.7 don't have importlib as part of their
+standard library, but it is possible to install it separately. If you're
+using python 2.6 (which is what is available on CentOS 6, for example),
+you may need to run::
+
+  pip install importlib
+
+
+Setting up the Database:
+========================
+By default dev environment uses SQLite as a database, so if you're using it you can skip this section.
+
+If you wish to use postgreSQL instead, you may get an error ``psycopg2 package not found``.
+  
+When you do ``haas-admin db create`` in the next step, you may need to install 
+following package on your system 
+
+CentOS::  
+
+  yum install postgresql-devel
+
+Ubuntu::
+  
+  sudo apt-get install libpq-dev
+
+before installing ``psycopg2`` in the virtualenv for HaaS::
+
+  pip install psycopg2
+
+
+Configuring HaaS
+================
+
+Now the ``haas`` executable should be in your path.  First, create a
+configuration file ``haas.cfg``. There are two examples for you to work from,
+``examples/haas.cfg.dev-no-hardware``, which is oriented towards development, and
+``examples/haas.cfg`` which is more production oriented.  These config
+files are well commented; read them carefully.
+
+HaaS can be configured using ``haas.cfg`` to not perform state-changing operations on nodes,
+headnodes and networks, allowing developers to run and test parts of a haas
+server without requiring physical hardware. To suppress actual node and headnode
+operations, set ``dry_run = True`` in the ``[devel]`` section. 
+
+
+Most customization require including directives under section ``[extensions]``
+
+For suppressing actual network switch operations, use the ``mock`` switch driver :: 
+  haas.ext.switches.mock =
+
+You can choose to disable authentication mechanism by setting::
+  haas.ext.auth.null =
+
+To enable authentication mechanism, set appropriate authentication backend.
+Authentication directives are mutually exclusive. To choose database as an 
+authentication backend::
+  haas.ext.auth.database =
+
+
+Next initialize the database with the required tables::
+
+  haas-admin db create
+  
+Run the server with the port number as defined in ``haas.cfg``
+
+  haas serve <port no> 
+  
+and in a separate window terminal
+
+  haas serve_networks
+  
+Finally, ``haas help`` lists the various API commands one can use.
+Here is an example session, testing ``headnode_delete_hnic``::
+
+  haas project_create proj
+  haas headnode_create hn proj
+  haas headnode_create_hnic hn hn-eth0
+  haas headnode_delete_hnic hn hn-eth0
+
+Additionally, before each commit, run the automated test suite with ``py.test
```

Comment:
We have detailed explanation in `developer guidelines`, but since this instructions were here already, I felt no harm in it getting repeated a little bit. It is good conditioning for new developers.